pyvista/core/cell.py

```diff
@@ -416,6 +424,21 @@ def _get_attrs(self):
 
         return attrs
 
+    def __eq__(self, other):
+        """Cell equality."""
+        # cells must have the same points and point ids
+        if not isinstance(other, Cell):
+            return False
+        if not self.type == other.type:
+            return False
+        if not np.array_equal(self.points, other.points):
+            return False
+        if not np.array_equal(self.point_ids, other.point_ids):
+            return False
+        if self.type == CellType.POLYHEDRON:
+            return all(zip(self.faces, other.faces))
```

Comment:
```suggestion
            # POLYHEDRON has a nested structure that depends on the faces
            return all(zip(self.faces, other.faces))
```

Reply:
`all(zip(self.faces, other.faces))` looks very surprising here. Did you not mean to compare pairs of values instead? And is it guaranteed at this point that the faces attributes have the same lengths? Otherwise we'd need `itertools.zip_longest()` or an explicit check.

Reply:
Actually, yeah, since `zip` always yields 2-tuples, and in the empty case we get a vacuous truth, this line always returned `True` I think.

Reply:
I think you are right, but this check failed on checking point_ids when shallow copying cells.  This is a rabbit hole I didn't want to go down for this PR

Reply:
I just wasn't sure whether you were going to reuse this somewhere else :) All good.